utils/auth/jwtHelpers.ts

```diff
@@ -0,0 +1,105 @@
+import admin from 'firebase-admin';
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+
+// Note: do not modify the 'firebase-admin' default import style, it triggers a known bug
+// See: https://github.com/firebase/firebase-admin-node/issues/593
+try {
+  admin.initializeApp({
+    projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
+  });
+} catch (error) {
+  // Ignore the "already exists" message in dev environments (cause by hot reloads).
+  if (!(process.env.NODE_ENV == 'development' && /already exists/u.test('' + error))) {
+    console.error(error);
+  }
+}
+
+const UNABLE_TO_AUTHENTICATE = 'User could not be authenticated.';
+const ANONYMOUS_USER_NOT_AUTHORIZED = 'User has not verified their identity.';
+
+/**
+ * Utility function that takes care of authenticating any user (anonymous / verified) and providing the UID.
+ *
+ * Only calls the provided handler if the user has been successfully authenticated. Returns a 401 Unauthorized response
+ * if requirements are not met.
+ *
+ * @param handler Takes in an API handler function that needs an authenticated user.
+ * @returns NextApiHandler function.
+ */
+export function withAnyUser<T>(
+  handler: (uid: string, req: NextApiRequest, res: NextApiResponse<T>) => void,
+): NextApiHandler<T> {
+  return async (req: NextApiRequest, res: NextApiResponse<T>) => {
+    let userDetails: UserDetailsFromRequest;
+    try {
+      userDetails = await getUserFromJwt(getBearerToken(req));
+    } catch (e) {
+      return res.status(401).end(UNABLE_TO_AUTHENTICATE);
+    }
+
+    return handler(userDetails.uid, req, res);
+  };
+}
+
+/**
+ * Utility function that takes care of authenticating a verified user and providing the UID.
+ *
+ * Only calls the provided handler if the user has been successfully authenticated and is verified. Returns a 401
+ * Unauthorized response if requirements are not met.
+ *
+ * @param handler Takes in an API handler function that needs a verified user.
+ * @returns NextApiHandler function.
+ */
+export function withVerifiedUser<T>(
+  handler: (uid: string, req: NextApiRequest, res: NextApiResponse<T>) => void,
+): NextApiHandler<T> {
+  return async (req: NextApiRequest, res: NextApiResponse<T>) => {
+    let userDetails: UserDetailsFromRequest;
+    try {
+      userDetails = await getUserFromJwt(getBearerToken(req));
+    } catch (e) {
+      return res.status(401).end(UNABLE_TO_AUTHENTICATE);
+    }
+
+    const { uid, isAnonymous } = userDetails;
+    if (isAnonymous) {
+      return res.status(401).end(ANONYMOUS_USER_NOT_AUTHORIZED);
+    }
+
+    return handler(uid, req, res);
+  };
+}
+
+interface UserDetailsFromRequest {
+  uid: string;
+  isAnonymous: boolean;
+}
```

Comment:
Should we move types and interfaces to the top of the file or a separate file? For easy finding. Right now it's used both in the function above and below it.